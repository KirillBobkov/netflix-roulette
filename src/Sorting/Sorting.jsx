import React from 'react';
import { Button } from '../primitives';
import './Sorting.scss';

export default function Sorting() {
  return (
    <div className='sorting'>
        <div className='sorting__container'>
            <span className='sorting__sort-description'>Sort by</span>
            <Button 
            className='button button--choosen button--left-border'
            text={'Release date'} 
            />
            <Button 
            className='button button--right-border'
            text={'Rating'} 
            />      
        </div>  
    </div>
  );
}



