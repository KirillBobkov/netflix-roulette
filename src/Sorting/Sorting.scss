    
    .sorting {
        background-color: #555555;

    
        &__container {
            width: 770px;
            padding: 15px 0;
            margin: 0 auto;
            text-align: right;
        }

        &__sort-description {
            display: inline-block;
            margin-right: 10px;
            font-size: 12px;
            text-transform: uppercase;
            color: #ffffff;
        }
    }

    