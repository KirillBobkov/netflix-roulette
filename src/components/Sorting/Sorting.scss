    
    .sorting {
        background-color: #555555;

    
        &__container {
            width: 770px;
            padding: 15px 0;
            margin: 0 auto;
            display: flex;
            justify-content: space-between;
        }

        &__sort-description {
            display: inline-block;
            margin-right: 24px;
            font-size: 12px;
            text-transform: uppercase;
            color: #ffffff;
        }

        &__match-movies {
            font-size: 12px;
            line-height: 12px;
            font-family: 'Monseratt-Bold', sans-serif;
            align-self: center;
        }
    }

    