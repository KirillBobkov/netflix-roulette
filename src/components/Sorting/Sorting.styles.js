import styled from "styled-components";

export const SortingWrapper = styled.div`
    background-color: #555555;
    width: 100%;
`;

export const SortingContainer = styled.div`
    width: 770px;
    padding: 15px 0;
    margin: 0 auto;
    display: flex;
    justify-content: space-between;
`;

export const SortingDescription = styled.span`
    display: inline-block;
    margin-right: 24px;
    font-size: 12px;
    text-transform: uppercase;
    color: #ffffff;
`;

export const SortingMatchMovies = styled.span`
    font-size: 12px;
    line-height: 12px;
    font-family: 'Monseratt-Bold', sans-serif;
    align-self: center;
`;


    