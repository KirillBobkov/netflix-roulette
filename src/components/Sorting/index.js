export { default as Sorting } from './SortingContainer';
