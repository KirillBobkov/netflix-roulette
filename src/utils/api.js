import axios from 'axios';

const api = axios.create({
    baseURL: "https://reactjs-cdp.herokuapp.com/",
    responseType: "json"
});

export const getMovies = (params = { 
    sortBy: "release_date", 
    sortOrder: "asc", 
    searchBy: "title" 
}) => {
    return api.get('/movies', {
        params
    });
};

export const getImage = url => {
    return new Promise((resolve, reject) =>{
        var img = new Image();
        img.onload = function(){
            resolve(url);
        };
        img.onerror = function(){
            reject(url);
        };
        img.src = url;
    });
};




