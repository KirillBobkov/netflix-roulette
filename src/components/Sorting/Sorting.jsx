import React from 'react';
import { Button } from '../primitives';
import './Sorting.scss';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { fetchMovies } from '../../store/actions/movieActions';
import { withRouter } from 'react-router-dom';

class Sorting extends React.PureComponent {

  handleSortBy = () => {
    const { fetchMovies, filter: { searchBy, search, sortBy } } = this.props;
    let sortByParam = sortBy === "release_date" ? "vote_average" : "release_date";

    // setMovies({
    //     sortBy: sortByParam,
    //     sortOrder: "asc",
    //     searchBy,
    //     search,
    //   });

    fetchMovies();
  }

  renderMoviePageUI() {
    const { list, filter: { searchBy } } = this.props;

    return (
      <div className='sorting'>
        <div className='sorting__container'>
          <span className='sorting__match-movies'>Filtered by {searchBy}</span>
        </div>
      </div>
    );
  }

  renderMainPageUI() {
    const { list, filter: { sortBy } } = this.props;
    const sortByDate = sortBy === "release_date";
    const dateClassName = sortByDate ? 'button--choosen' : '';
    const ratingClassName = !sortByDate ? 'button--choosen' : '';

    return (
      <div className='sorting'>
        <div className='sorting__container'>
          <span className='sorting__match-movies'>{list && list.length} movies found</span>

          <p>
            <span className='sorting__sort-description'>Sort by</span>
            <Button
              className={`${dateClassName} button--left-border`}
              text='Release date'
              onClick={this.handleSortBy}
            />
            <Button
              className={`${ratingClassName} button--right-border`}
              text='Rating'
              onClick={this.handleSortBy}
            />
          </p>
        </div>
      </div>
    );
  }

  render() {
    const { isMoviePage } = this.props;
  
    return isMoviePage
    ? this.renderMoviePageUI()
    : this.renderMainPageUI();
  }
}

Sorting.propTypes = {
  fetchMovies: PropTypes.func,
  isMoviePage: PropTypes.bool,
  list:  PropTypes.array,
  filter: PropTypes.shape({
    sortBy: PropTypes.string,
    searchBy: PropTypes.string,
    search: PropTypes.string
  })
};

const mapStateToProps = (state, ownProps) => { 
  const { list, filter } = state;

  return { 
    list, 
    filter, 
    isMoviePage: Boolean(ownProps.match.params.id) };
};

const mapDispatchToProps = dispatch => ({
  fetchMovies: () => dispatch(fetchMovies())
});

export default withRouter( connect(mapStateToProps, mapDispatchToProps)(Sorting) );

