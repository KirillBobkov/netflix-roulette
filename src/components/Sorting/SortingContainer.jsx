import { connect } from 'react-redux';
import { fetchMovies } from '../../store/actions/movieActions';
import { withRouter } from 'react-router-dom';
import { Sorting } from './Sorting';

const mapStateToProps = (state, ownProps) => { 
  const { list, filter } = state;

  return { 
    list, 
    filter, 
    isMoviePage: Boolean(ownProps.match.params.id) };
};

const mapDispatchToProps = dispatch => ({
  fetchMovies: (config) => dispatch(fetchMovies(config))
});

export default withRouter( connect(mapStateToProps, mapDispatchToProps)(Sorting) );
