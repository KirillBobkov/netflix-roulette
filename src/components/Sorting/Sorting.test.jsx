import React from 'react';
import Sorting from './Sorting';
import { getMovies } from '../../utils';
import { Provider } from 'react-redux';
import { movies } from '../../utils/testData';
import configureStore from 'redux-mock-store';
import { shallow, mount } from 'enzyme';


describe('Sorting component', () => {
    const initialState = movies;
    const mockStore = configureStore();
    const mountWithStore = component => mount(<Provider store={store}> {component} </Provider>);
    let store, 
        wrapperSorting;
    
    beforeAll(()=>{
      store = mockStore(initialState);
      wrapperSorting = mountWithStore(<Sorting />);
    });
    
    it('should be render correctly', () => {
        expect(wrapperSorting).toMatchSnapshot();
    });

    it('should render title of Sorting', () => {
        expect(wrapperSorting.exists('.button')).toEqual(true);
    }); 

    it('should render sorting description', () => {
        const sortDescription = wrapperSorting.find('.sorting__sort-description');
        expect(sortDescription.text()).toEqual('Sort by');
    });
});    



