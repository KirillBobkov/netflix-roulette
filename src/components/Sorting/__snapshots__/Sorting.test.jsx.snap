// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sorting component should be render correctly 1`] = `ShallowWrapper {}`;
